event/helper.go

```diff
@@ -0,0 +1,121 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package event
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func multipleObjectEvents(ctx context.Context, m Manager, objects []types.ManagedObjectReference, pageSize int, tail bool, force bool, prop []string, f func([]types.BaseEvent) error) error {
+	// create an EventHistoryCollector for each object
+	var collectors []types.ManagedObjectReference
+	for _, o := range objects {
+		filter := types.EventFilterSpec{
+			Entity: &types.EventFilterSpecByEntity{
+				Entity:    o,
+				Recursion: types.EventFilterSpecRecursionOptionAll,
+			},
+		}
+
+		collector, err := m.CreateCollectorForEvents(ctx, filter)
+		if err != nil {
+			return fmt.Errorf("[%#v] %s", o, err)
+		}
+		defer collector.Destroy(ctx)
+
+		err = collector.SetPageSize(ctx, pageSize)
+		if err != nil {
+			return err
+		}
+
+		collectors = append(collectors, collector.Reference())
+	}
+
+	// create and populate a ListView
+	viewMgr := view.NewManager(m.c)
+	listView, err := viewMgr.CreateListView(ctx, collectors)
+	if err != nil {
+		return err
+	}
+	// Retrieve the property from the objects in the ListView
+	count := 0
+	err = property.WaitForView(ctx, property.DefaultCollector(m.c), listView.Reference(), collectors[0], prop, func(pc []types.PropertyChange) bool {
+		for _, u := range pc {
+			count++
+			if u.Name != prop[0] {
+				continue
+			}
+			if u.Val == nil {
+				continue
+			}
+			f(u.Val.(types.ArrayOfEvent).Event)
+			if count == len(collectors) && !tail {
+				return true
+			}
+
+		}
+		return false
+	})
+
+	return nil
+}
+
+func singleObjectEvents(ctx context.Context, m Manager, object types.ManagedObjectReference, pageSize int, tail bool, force bool, prop []string, f func([]types.BaseEvent) error) error {
+	filter := types.EventFilterSpec{
+		Entity: &types.EventFilterSpecByEntity{
+			Entity:    object,
+			Recursion: types.EventFilterSpecRecursionOptionAll,
+		},
+	}
+
+	collector, err := m.CreateCollectorForEvents(ctx, filter)
+	if err != nil {
+		return fmt.Errorf("[%#v] %s", object, err)
+	}
+	defer collector.Destroy(ctx)
+
+	err = collector.SetPageSize(ctx, pageSize)
+	if err != nil {
+		return err
+	}
+
+	count := 0
+	err = property.Wait(ctx, property.DefaultCollector(m.c), collector.Reference(), prop, func(pc []types.PropertyChange) bool {
+		for _, u := range pc {
+			count++
+			if u.Name != prop[0] {
+				continue
+			}
+			if u.Val == nil {
+				continue
+			}
+			f(u.Val.(types.ArrayOfEvent).Event)
+			if !tail {
```

Comment:
I never saw that in testing (exiting the loop early), but after reviewing the code there as an issue waiting to happen...specifically there's no need to count the objects in the multiple method...we're either tailing or we aren't...anyhoo...moved the check outside of the loop and simplified a bit